### PR TITLE
Welders no longer spam eye flashes

### PR DESCRIPTION
**Pull request by Citinited**
**What does this PR do:**

Tweaks the change added in #11700 for welders. Instead of spamming `flash_eyes()` every second, instead now it just adds the flash overlay every 2.5 seconds (which essentially makes your screen white all the time while welding). It also won't give you more and more damage over time.

**Obligatory picture:**
image

:cl:
tweak: Welders now apply the flash overlay without a copious amount of eye damage.
/:cl:

### CODE REVIEW

**Comment by Darkmight9** [comment]
oh god yes please!

File: code/game/objects/items/tools/welder.dm [hunk 1]
```diff
@@ -122,8 +122,8 @@
 /obj/item/weldingtool/tool_check_callback(mob/living/user, amount, datum/callback/extra_checks)
 	. = ..()
 	if(. && user)
-		if(progress_flash_divisor == 0)
-			user.flash_eyes(min(light_intensity, 1))
+		if(progress_flash_divisor <= 0)
+			user.add_flash_overlay(min(light_intensity, 1))
```
Comment (Crazylemon64) [inline code/game/objects/items/tools/welder.dm:126]: flash_eyes checked for eye protection before doing the flash - this doesn't

**Comment by Crazylemon64** [comment]
The PR is OK design-wise, but it should flash only when eye protection is insufficient

File: code/game/objects/items/tools/welder.dm (hunk 1 above)
Comment (farie82) [reply code/game/objects/items/tools/welder.dm:126]: He moved the protection check to add_flash_overlay and made flash_eyes call add_flash_overlay as a base.
So the protection check will be done but only the eye damage and other stuff flash_eyes might do will not be applied

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -756,11 +756,15 @@
 	return "You can't fit into that vent."
 
 //called when the mob receives a bright flash
-/mob/living/proc/flash_eyes(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type = /obj/screen/fullscreen/flash)
+/mob/living/proc/flash_eyes(intensity, override_blindness_check, affect_silicon, visual, type)
```
Comment (farie82) [inline code/modules/mob/living/living.dm:759]: The default values here are gone. Meaning all calls depending on those will break

Comment (Fox-McCloud) [reply code/modules/mob/living/living.dm:759]: Yeah, this needs to be addressed.

**Comment by Fox-McCloud** [comment]
Please deconflict and resolve the issues, please. This is quite stale.

**Comment by Fox-McCloud** [comment]
Nothing wrong with this PR, but closing due to staleness.

**Pull request closed without merging**